cmd/frontend/internal/httpapi/internal.go

```diff
@@ -113,6 +113,28 @@ func servePhabricatorRepoCreate(w http.ResponseWriter, r *http.Request) error {
 	return nil
 }
 
+func serveExternalServiceConfigs(w http.ResponseWriter, r *http.Request) error {
+	var req api.ExternalServiceConfigsRequest
+	err := json.NewDecoder(r.Body).Decode(&req)
+	if err != nil {
+		return err
+	}
+	services, err := db.ExternalServices.List(r.Context(), db.ExternalServicesListOptions{Kind: req.Kind})
+	if err != nil {
+		return err
+	}
+
+	var configs []map[string]interface{}
```

Comment:
We can avoid potential multiple heap allocations here because we know the upper bound of `len(configs)`.

```suggestion
configs := make([]map[string]interface{}, 0, len(services))
```